docs/source/plugins/for_plugin_developers.rst

```diff
@@ -0,0 +1,291 @@
+.. _plugins-for-plugin-developers:
+
+.. role:: python(code)
+   :language: python
+
+creating a napari plugin
+========================
+
+This document explains how to extend napari functionality by writing a plugin
+that can be installed with ``pip`` and autodetected by napari.  For in-depth
+information on how plugins are used internally in napari, see
+:ref:`plugins-for-napari-developers`.
+
+
+Overview
+--------
+
+``napari`` allows developers to extend the functionality of the program as
+follows:
+
+1. **Hooks**: We (napari) identify specific places ("*hooks*") in the napari
+codebase where we would like external developers to be able to extend
+functionality. For example, when a user tries to open a filepath in napari, we
+might want to enable plugins to extend the file formats that can be handled.  A
+*hook*, then, is the place within napari where we
+"promise" to call functions created by external developers.
+
+2. **Hook Specifications**:  We then create "*hook specifications*", which are
+well-documented function signatures that define the API (or
+"contract") that a plugin developer must adhere to when writing their function
+that we promise to call somewhere in the napari codebase.
+
+3. **Hook Implementations**: Plugin developers then write functions ("*hook
+implementations*") and mark that function as meeting the requirements of a
+specific *hook specification* offered by napari (using a decorator as
+:ref:`described below <hookimpl-decorator>`).
+
+4. **Plugin discovery**: Using :ref:`one of two methods <hookimpl-decorator>`,
```

Comment:
You mention "one of two methods" here, but don't say which they are, could you maybe say something and then have the link, otherwise I find myself wanting more information here

Reply:
Maybe just mention a "naming convention" or "package metadata"